### PR TITLE
New drops for zombie soldiers [CR]

### PR DESCRIPTION
**Pull request by John-Candlebury**
Creates a different and rather elaborate set of drops for the basic zombie soldiers. 

The main changes are: 
-They wear actual uniforms
-Normal army weapons  and ammo should become a bit more common.
-Drop damaged weapons
-Soldier zombies with weapons will  drop  bullets for the gun they are carrying. Usually less than a mag.

Might not be entirely balanced, as the whole thing was originally intended to place soldier corpses in a custom building I was working on. I changed  most chances to make them smaller, but they might still need adjusting.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
Very nice!

File: data/json/monster drops/zombie_soldier_death_drops.json [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    {
+        "type" : "item_group",
+        "subtype": "collection",
+        "id": "soldier_m14ebr",
+        "entries":
+        [
+            { "item": "spare_mag", "prob": 50, "count": 2},
+            { "item": "binoculars", "prob": 50},
+            { "item": "kevlar", "damage-min": 1, "damage-max": 4, "prob": 50},
+            { "item": "m14ebr", "charges-min": 0, "charges-max": 30, "ammo": "762_51"},
```
Comment (Coolthulhu) [inline data/json/monster drops/zombie_soldier_death_drops.json]: No damage here?

**Comment by Coolthulhu** [comment]
Ammo shouldn't be guaranteed on any of the drops - it's easiest to die when you run out.

**Comment by Coolthulhu** [comment]
Maybe also buff the soldiers stats a bit while we're at it? Armor especially, they have very unimpressive armor compared to what they drop.

**Comment by John-Candlebury** [comment]
I'm not sure if the zombies need much higher armor, they already have 16 cut. Which  means that the normal attacks of stuff like knives and machetes do almost no damage, with only crits dealing considerable damage (and that's because crits are generally mad op).

 Perhaps they could get some extra (+30 or so) health and some extra 5 movement points instead?

**Comment by kevingranade** [comment]
Knives and machetes should not be able to penetrate body armor, that's kind
of the point.
I have 0 problem with requiring heavier weapons to take on certain enemies,
especially if they're dropping good gear now.
Put another way, if the armor is that excessive, why are we giving it to
the player?

**Comment by John-Candlebury** [comment]
Gave them them more armor. 

>Put another way, if the armor is that excessive, why are we giving it to
the player?

Well deflection is chance based when it comes to the player, but a guaranteed reduction when it comes to monsters. So I'm not sure if giving  monsters  armor ratings that are  comparable to player armor is a good idea.

Although the player does have an  advantage when it comes to critical hits.

**Comment by Coolthulhu** [comment]
This REALLY makes them loot pinatas. It didn't look as good on paper, but actual results are:
* A single medic zombie pretty much sets you for life with all the medicine you could need. Mythical or not, immunity to infection for the rest of the game, average of 6 uses of the endgame painkillers, tramadol, first aid and bandages should not spawn all on one corpse
* Good guns with respectable ammo stacks are common - mostly 5.56, which has really good stats even when fired from a damaged gun
* Handguns are undamaged on spawn
* For some reason, they spawn with much more 40mm grenades than launchers

It's mostly the medics and ammo.
They could get more hand grenades to compensate. Those are quite rare.

**Pull request merged**